packages/block-editor/src/hooks/layout.js

```diff
@@ -405,26 +403,23 @@ export const withLayoutStyles = createHigherOrderComponent(
 			layoutClasses
 		);
 
-		return (
-			<>
-				{ shouldRenderLayoutStyles &&
-					element &&
-					!! css &&
-					createPortal(
-						<LayoutStyle
-							blockName={ name }
-							selector={ selector }
-							css={ css }
-							layout={ usedLayout }
-							style={ attributes?.style }
-						/>,
-						element
-					) }
-				<BlockListBlock
-					{ ...props }
-					__unstableLayoutClassNames={ layoutClassNames }
+		BlockList.useRootPortal(
+			shouldRenderLayoutStyles && !! css && (
+				<LayoutStyle
+					blockName={ name }
+					selector={ selector }
+					css={ css }
+					layout={ usedLayout }
+					style={ attributes?.style }
 				/>
-			</>
+			)
+		);
+
+		return (
+			<BlockListBlock
+				{ ...props }
+				__unstableLayoutClassNames={ layoutClassNames }
+			/>
```

Comment:
To do: use the new filter here too.

Reply:
This maybe needs to be a `useInnerBlocksProps` instead of a `useBlockProps` filter.